packages/components/form/__specs__/phone_input.spec.tsx

```diff
@@ -0,0 +1,89 @@
+import { useForm } from "react-hook-form";
+import { fireEvent } from "@testing-library/react";
+
+import PhoneInput, { PhoneInputProps } from "../src/phone_field/phone_input";
+import { Form } from "../src/form";
+import { render, UserRenderResult } from "../test_utils";
+
+const BACKSPACE_KEY = {
+  key: "Backspace",
+  code: 8,
+  inputType: "deleteContentBackward",
+};
+
+const SHARED_EVENT_CONFIG = {
+  key: BACKSPACE_KEY.key,
+  charCode: BACKSPACE_KEY.code,
+  keyCode: BACKSPACE_KEY.code,
+  which: BACKSPACE_KEY.code,
+};
+
+describe("<PhoneInput />", () => {
+  type MockFormProps = Partial<PhoneInputProps>;
+
+  const backspace = (field: Element): void => {
+    let { value } = field as HTMLInputElement;
+
+    const downEvent = fireEvent.keyDown(field, SHARED_EVENT_CONFIG);
+
+    if (downEvent) {
+      value = value.slice(0, -1);
+
+      fireEvent.input(field, {
+        target: { value },
+        inputType: BACKSPACE_KEY.inputType,
+        bubbles: true,
+        cancelable: true,
+      });
```

Comment:
We've been utilizing UserEvent as recommended in our guides in anchor.

https://testing-library.com/docs/user-event/intro/

Reply:
My mistake

Reply:
Holy cow this is WAY better. 😂

Reply:
![](https://y.yarn.co/7ca981c6-4bdf-4dee-97b6-7a0f4141f4d1_text.gif)